Extension/src/LanguageServer/configurations.ts

```diff
@@ -186,13 +186,23 @@ export class CppProperties {
         this.defaultIncludes = compilerDefaults.includes;
         this.defaultFrameworks = compilerDefaults.frameworks;
 
-        if (this.defaultCompilerPath && this.defaultCompilerPath.length > 0) {
-            let config: Configuration = this.configurationJson.configurations[this.CurrentConfiguration];
-            if (!config.compilerPath || config.compilerPath.length === 0)
-            {
-                config.compilerPath = this.defaultCompilerPath; // Update the compilerPath with the default being used.
-                fs.writeFileSync(this.propertiesFile.fsPath, JSON.stringify(this.configurationJson, null, 4));
-            }
+        // Update the compilerPath, cStandard, and cppStandard with the default being used.
+        let doUpdate: boolean = false;
+        let config: Configuration = this.configurationJson.configurations[this.CurrentConfiguration];
+        if (this.defaultCompilerPath && this.defaultCompilerPath.length > 0 && (!config.compilerPath || config.compilerPath.length === 0)) {
```

Comment:
The .length checks should be unnecessary.  If this.defaultCompilerPath evaluates to true it can't be the empty string.